### PR TITLE
Yuwei/bugfix 202209

### PR DESCRIPTION
**Pull request by ivywei0125**
What does this implement/fix? Explain your changes.
---------------------------------------------------
1. update message when nothing found in subscriptions dialog. [AB#1983740](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1983740)
2. update UI of web app creation dialog to keep consistent with function app [AB#1983474](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1983474)
3. validate application insights name when entering character 
 [AB#1980248](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1980248)
4. update current artifact when open configuration dialog 
 [AB#1983480](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1983480)
5.  create new resource group in Get Started [AB#1983484](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1983484)
6. update current artifact when open configuration dialog  [AB#1983480](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1983480) 
7. draft should be available for all regions and all os. 
 [AB#1983441](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1983441)
8. update message [AB#1948968](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?text=insigh&workitem=1948968)
9. update validator for minNodeCount [AB#1971284](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1971284)
10. add validator for vm configuration [AB#1934588](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1934588)
11. disable "stop streaming log" action for linux function app [AB#1975294](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1975294)
12. clear web app in configuration if resource not found [AB#1982651](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1982651)
13. show waring notification if user try to download file from stopped app [AB#1972398](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1972398)
14. update enabled condition to display action view as expected [AB#1966288](https://dev.azure.com/mseng/VSJava/_workitems/edit/1966288)
15. update MinimumAndPreferredWidth to avoid combobox width change after setItems  [AB#1980245](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1980245)
16. add listener to tool window [AB#1968507](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?System.AssignedTo=%40me&workitem=1968507)
 

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/ui/SubscriptionsDialog.java [hunk 1]
```diff
@@ -146,6 +146,9 @@ private synchronized void updateTableView() {
         for (final SimpleSubscription sd : subs) {
             model.addRow(new Object[]{sd.isSelected(), sd.getName(), sd});
         }
+        if (model.getRowCount() <= 0) {
+            table.getEmptyText().setText("No subscriptions with such name found");
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/ui/SubscriptionsDialog.java]: -> No subscriptions

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/webapp/runner/webappconfig/slimui/WebAppDeployConfigurationPanel.java [hunk 2]
```diff
@@ -251,6 +252,10 @@ public void setValue(WebAppDeployRunConfigurationModel data) {
                 .ifPresent(artifact -> comboBoxArtifact.setArtifact(artifact));
         // web app
         Optional.ofNullable(data.getWebAppConfig()).ifPresent(webApp -> {
+            if (!Azure.az(AzureAccount.class).account().getTenantIds().contains(webApp.getSubscriptionId())) {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/webapp/runner/webappconfig/slimui/WebAppDeployConfigurationPanel.java]: tenant id is different from subscription id.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-arm/src/main/java/com/microsoft/azure/toolkit/intellij/arm/creation/CreateDeploymentDialog.java [hunk 3]
```diff
@@ -53,6 +53,7 @@ public class CreateDeploymentDialog extends AzureDialogWrapper {
 
     private static final String DUPLICATED_DEPLOYMENT_NAME = "A deployment with the same name already exists";
     private final Project project;
+    private final ResourceGroup initResourceGroup;
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-arm/src/main/java/com/microsoft/azure/toolkit/intellij/arm/creation/CreateDeploymentDialog.java]: `initialResourceGroup`

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/creation/VMCreationDialog.java [hunk 4]
```diff
@@ -359,6 +363,18 @@ private void createUIComponents() {
         this.cbSubscription.reloadItems();
     }
 
+    @Override
+    protected List<ValidationInfo> doValidateAll() {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/creation/VMCreationDialog.java:367]: also validate when the other is modified.

**Comment by wangmingliang-ms** [comment]
please fix npe in later commits.

**Pull request merged**